### PR TITLE
Extend README.md with Installation Guide and Documentation

### PR DESCRIPTION
**Pull request by juanlofer-eprosima**
Signed-off-by: Juan López Fernández <juanlopez@eprosima.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,3 +37,187 @@ Following are some of the key features of *eProsima DDS Router*:
...
+
+```batch
+-DCMAKE_INSTALL_PREFIX=/usr/local/ -DBUILD_SHARED_LIBS=ON
+```
+
+---
+**Note**
+
+By default, *DDS Router* does not compile tests. However, they can be activated by downloading and installing
+[Gtest](https://github.com/google/googletest) and building with CMake option `-DBUILD_TESTS=ON`.
```
Comment (jparisu) [inline README.md:210]: Add command to execute them.

File: README.md [hunk 2]
```diff
@@ -37,3 +37,187 @@ Following are some of the key features of *eProsima DDS Router*:
...
+
+To install *DDS Router* system-wide instead of locally, remove all the flags that appear in the configuration steps of `Fast-CDR`, `Fast-DDS`, and `DDS-Router`, and change the first in the configuration step of `foonathan_memory_vendor` to
+the following:
+
+```batch
+-DCMAKE_INSTALL_PREFIX=/usr/local/ -DBUILD_SHARED_LIBS=ON
+```
+
+---
+**Note**
```
Comment (jparisu) [inline README.md:207]: Maybe move it to a new subsection *Test*

File: README.md [hunk 3]
```diff
@@ -37,3 +37,187 @@ Following are some of the key features of *eProsima DDS Router*:
...
+vcs import src < ddsrouter.repos
+```
+
+2. Build the packages:
+
+```batch
+colcon build
+```
+
+### CMake installation
```
Comment (jparisu) [inline README.md:143]: NIT: I would not add this section. I do not think people use it, and If someone is interested, could go to the docu. I would do this Readme as a very easy and simple way to install, and not add extra possibilities

File: README.md [hunk 4]
```diff
@@ -37,3 +37,187 @@ Following are some of the key features of *eProsima DDS Router*:
...
+#### Gtest
+
+[Gtest](https://github.com/google/googletest) is a unit testing library for C++. By default, *DDS Router* does not
+compile tests. It is possible to activate them with the opportune [CMake options](https://colcon.readthedocs.io/en/released/reference/verb/build.html#cmake-options) when calling [colcon](https://colcon.readthedocs.io/en/released/) or
+[CMake](https://cmake.org/). For a detailed description of the Gtest installation process, please refer to the
+[Gtest Installation Guide](https://github.com/google/googletest).
+
+### Dependencies
+
+#### Asio and TinyXML2 libraries
```
Comment (jparisu) [inline README.md:97]: Add `yaml-cpp`

File: README.md [hunk 5]
```diff
@@ -37,3 +37,187 @@ Following are some of the key features of *eProsima DDS Router*:
...
+* [Use Cases](https://eprosima-dds-router.readthedocs.io/en/latest/rst/use_cases/ros_cloud.html)
+* [Developer Manual](https://eprosima-dds-router.readthedocs.io/en/latest/rst/developer_manual/installation/sources/linux.html)
+* [Release Notes](https://eprosima-dds-router.readthedocs.io/en/latest/rst/notes/notes.html)
+
+
+## Installation Guide
+
+The instructions for installing the *DDS Router* application from sources and its required dependencies on a Linux
+environment are provided below. These installation instructions are a summarized version of the complete
+[installation guide](https://eprosima-dds-router.readthedocs.io/en/latest/rst/developer_manual/installation/sources/linux.html) available online, which also includes the [required steps](https://eprosima-dds-router.readthedocs.io/en/latest/rst/developer_manual/installation/sources/windows.html) for installing *DDS Router* on a Windows platform.
```
Comment (jparisu) [inline README.md:59]:
```suggestion
[installation guide](https://eprosima-dds-router.readthedocs.io/en/latest/rst/developer_manual/installation/sources/linux.html) available online.
For installing *DDS Router* on a **Windows platform** follow next link https://eprosima-dds-router.readthedocs.io/en/latest/rst/developer_manual/installation/sources/windows.html .
```

File: README.md (hunk 4 above)
Comment (jparisu) [inline README.md:97]: NIT: separate between fastdds dependencies and router dependencies

File: README.md [hunk 6]
```diff
@@ -37,3 +37,187 @@ Following are some of the key features of *eProsima DDS Router*:
...
+
+### Run an application
+
+To run the *DDS Router* application, source the *Fast DDS* library and execute the executable file that has been
+installed in `<install-path>/ddsrouter/bin/ddsrouter`:
+
+```batch
+# If built has been done using colcon, all projects could be sourced as follows:
+source <install-path>/setup.bash
+./<install-path>/ddsrouter/bin/ddsrouter
```
Comment (jparisu) [inline README.md:222]:
```suggestion
# Source installation
source <install-path>/setup.bash

# Execute DDS Router
./<install-path>/ddsrouter/bin/ddsrouter
```

**Review by jparisu — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (juanlofer-eprosima) [reply README.md:98]: Also added it to Linux and Windows documentation.

File: README.md (hunk 1 above)
Comment (jparisu) [reply README.md]: add `--packages-select ddsrouter`, and maybe `--event-handler=console_direct+`

File: README.md (hunk 5 above)
Comment (jparisu) [reply README.md]: NIT: set Windows as bold, so it is easier to find?

File: README.md [hunk 7]
```diff
@@ -37,3 +37,143 @@ Following are some of the key features of *eProsima DDS Router*:
...
+
+```batch
+sudo apt install cmake g++ pip wget git
+```
+
+#### Colcon
+
+[colcon](https://colcon.readthedocs.io/en/released/) is a command line tool based on [CMake](https://cmake.org/) aimed at building sets of software packages. Install the ROS 2 development tools ([colcon](https://colcon.readthedocs.io/en/released/) and [vcstool](https://pypi.org/project/vcstool/)) by executing the following command:
+
+```batch
```
Comment (jparisu) [inline README.md:83]: NIT:
In general:
Use `bash` or `sh`, not `batch`.
```suggestion
```bash
```

File: docs/rst/developer_manual/installation/sources/windows.rst [hunk 8]
```diff
@@ -172,6 +173,22 @@ For example:
...
+yaml-cpp is a YAML parser and emitter in C++ matching the YAML 1.2 spec, and is used by *DDS Router* application to
+parse the provided configuration files.
+From an administrative shell with *PowerShell*, execute the following commands in order to download and install yaml-cpp
+for Windows:
+
+.. code-block:: bash
+
+   git clone --branch yaml-cpp-0.7.0 https://github.com/jbeder/yaml-cpp
+   cmake -DCMAKE_INSTALL_PREFIX='C:\Program Files\yamlcpp' -B build\yamlcpp yaml-cpp
+   cmake --build build\yamlcpp --target install
```
Comment (jparisu) [inline docs/rst/developer_manual/installation/sources/windows.rst:190]: Add CMakeBuildType and say it needs to be the same as the ddsrouter type

**Review by jparisu — CHANGES_REQUESTED** [review]

**Review by jparisu — APPROVED** [review]
LGTM

**Review by jparisu — APPROVED** [review]
LGTM

**Pull request merged**